main.rb

```diff
@@ -0,0 +1,118 @@
+module Enumerable
+  
+  # my each
+  def my_each
+    count = 0
+    while count < length
+      yield(self[count])
+      count += 1
+    end
+    self
+  end
+  
+  # my_each_with_each
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+  
+  # my_select
+  def my_select
+    count = 0
+    arr = []
+    while count < length
+      arr.push(self[count]) if yield(self[count])
+      count += 1
+    end
+    arr
+  end
+  
+  # my_all
+  def my_all?
+    count = 0
+    while count < length
+      count += 1
+      return false unless yield(self[count])
+      
+    end
+    true
+  end
+  
+  # my_any?
+  def my_any?
```

Comment:
We have another issue here. The index, as in the past comment, is starting at 1 not 0.
With the same test case as the previous comment this block`[ 1, 1, 1, 1, 1 ].my_any? { |n| n < 1}  does not return false as it should, it returns an error. Kindly see that it replicates the same functionality as the `any` ruby enumerable.